src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,428 @@
+      return new HtmlResponse('No instanceID found.', 500);
+    }
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    $response = [
+      '#theme' => 'eloqua_app_cloud_update',
+    ];
+    // Punting on the case where there are multiple plugins for a service.
+    // This code will always return all the instantiate responses for the
+    // last plugin in the list, but does not try to help if they break each other.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      $update = $plugin->update($instanceId, $query);
+      $response['#content'][$plugin->getPluginId()] = $update;
+    }
+    return $response;
+  }
+
+  /**
+   * Delete any existing queue entries for this service entity.
+   *
+   * @TODO: Call the plugins deete method to let it do specific cleanup instead
+   *   of doing it here.
+   * @return mixed
+   */
+  public function delete($eloqua_app_cloud_service) {
+    $query = $this->request->query->all();
+    // Get the instanceID from the query parameter.
+    $instanceId = $query["instance"];
+    if (empty($instanceId)) {
+      $this->logger->error('No instanceID found');
+      // It apparently does not matter what we return here, since Eloqua does not have any way of handing external errors?
+      return new HtmlResponse('No instanceID found.', 500);
+    }
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    // Iterate over the ServiceEntity plugins and then over the payload items.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+
+      // Get the appropriate queue for this plugin.
+      $queue = $this->queueFactory->get($plugin->queueWorker());
+      $queueCount = $queue->numberOfItems();
+
+      for ($i = 0; $i <= $queueCount; $i++) {
+        $queueItem = $queue->claimItem();
+        // Only delete the item if it is part of our instance.
+        // If the claim returned false then assume the queue is empty.
+        if ($queueItem && $queueItem->instanceId === $instanceId) {
+          $queue->deleteItem($queueItem);
+        }
+      }
+    }
+  }
+
+  /**
+   * Execute.
+   *
+   * @return mixed
+   */
+  public function execute($eloqua_app_cloud_service) {
+    $query = $this->request->query->all();
+    // Get the instanceID from the query parameter.
+    $instanceId = $query["instance"];
+    if (empty($instanceId)) {
```

Comment:
- [ ] Noticing this coming up in basically all of these.  Consider refactoring into an additional [access check](https://www.drupal.org/docs/8/api/routing-system/access-checking-on-routes) on all of the controller routes?  Or at least a private method.